packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -415,13 +451,10 @@ export const useUpdate = <RecordType extends RaRecord = any, ErrorType = Error>(
 
         if (mode.current === 'optimistic') {
             // call the mutate method without success side effects
-            return mutation.mutate(
-                { resource: callTimeResource, ...callTimeParams },
-                {
-                    onSettled: otherCallTimeOptions.onSettled,
-                    onError: otherCallTimeOptions.onError,
-                }
-            );
+            return mutation.mutate({
+                resource: callTimeResource,
+                ...callTimeParams,
```

Comment:
The onSettled and onError side effects have disappeared here. I assume it's because you've set the refs and they are called in the definition-time side effects. If that's the case, please add more comments here to help remember it.

Reply:
bump